### DISCUSSION TITLE
Current state of v2 profiles

### DISCUSSION THREAD

**Discussion by MarvinJWendt** [creativeprojects/resticprofile#345] [Q&A] [answered]
Hi, first of all, thanks for the awesome project, this is what restic needs!

### Intro

I am currently using Duplicati to backup multiple servers, laptops and desktops.
I want to migrate to restic, as Duplicati is too unmaintained for my taste.

I have read trough the v2 profile proposal and I really like the new configuration scheme.
But since I have a lot of configured backups, I don't want to migrate those too often.

### My Question(s)

1. Is there any ETA for the v2 profiles?
	- I know giving ETAs on open source projects is very difficult (I maintain some projects myself), but since v2 profiles are in discussion since the end of 2021, I just want to know, if we can expect stable v2 profiles soon or in the far future.
2. How stable are v2 profiles?
	1. How stable is the current implementation?
	2. Should we expect to do frequent migrations of the profiles if we use v2 now? 
3. What is missing in the v2 profiles?
	1. Is there anything that cannot be done with v2 profiles, but with v1 profiles at the current point?
4. Will the migration from v1 to v2 be easy?
	1. Will there be a CLI command to do the migration of config files automatically?
6. Last but not least, would you recommend me to setup ~5 backup profiles across 3 machines with v1 or v2 profiles, at the current state?

Thanks in advance! :)

**Comment by creativeprojects** [comment, marked as answer]
1. We're nearly there actually. I'd say we're probably going to be finished in a few months.
2. We don't expect any more (big) changes at this stage. We already have a few v2 configurations in prod.
3. The last bit missing is the scheduling of group of profiles (scheduling to run multiple profiles one after another).
The only thing that is removed from v1 is the support for HCL configuration files.
4. I thought about it many times, but it's really not a big change. Basically between v1 and v2 you move all your profiles into a `profiles` subsection instead of having them defined at the root. After that you can start using `mixins` and other advanced features. 
5. Yes I think you can go directly onto v2 👍🏻

**Reply by MarvinJWendt** [reply]
Thanks for the answer!
